lib/collectors.js

```diff
@@ -2,6 +2,46 @@
 
 module.exports = (api) => {
 
+  api.metasync.collect = (expected) => {
+
+    const isCount = typeof(expected) === 'number';
+    const isKeys = Array.isArray(expected);
+    if (!(isCount || isKeys)) throw new Error('Unexpected type');
+    const keys = expected;
+    if (isKeys) expected = expected.length;
+    let count = 0;
+    let timeout = 0;
+    let timer = null;
+    let done = null;
+    const data = {};
+
+    const collector = (key, value) => {
+      if (isCount || keys.includes(key)) count++;
+      if (value instanceof Error) return done(value, data);
```

Comment:
@aqrln it's not a callback to use errback contract. `collector` is a function with signature `(key, value)`. We can change contract to `(key, error, value)` but it will not be compatible with errback.